### ISSUE TITLE
part_candidate coordinates problem

### ISSUE THREAD

**Issue by DongJT1996** [CMU-Perceptual-Computing-Lab/openpose#475]
### Issue Summary

I use the following command to save the body part candidates. As far as I know, body part candidates and PAFs are the input of the paper's optimization algorithm to get the final person keypoints. So in the output file, i.e. 1_keypoints.json, i think that the "pose_keypoints" is the final person keypoints and the 
"part_candidates" is the body part candidates. And i think the points in "pose_keypoints" is coming from "part_candidates". But the result is not like that, I can't find the points in "pose_keypoints" from "part_candidates". And when i plot the "part_candidates" in the input image, the result is wrong, the point is not in the place of joints. 

A bit wordy, but hopefully you can understand what I mean. 
thanks
### Executed Command (if any)

./build/examples/openpose/openpose.bin --image_dir examples/media/input/ --part_candidates --write_json output/heatmaps/5/

### OpenPose Output (if any)
1_keypoints.json

{"version":1.1,"people":[{"pose_keypoints":[435.167,210.626,0.856558,444.27,221.065,0.881522,431.21,223.624,0.876429,463.811,241.967,0.98562,486.009,228.905,0.595711,459.936,215.866,0.819104,479.497,214.519,0.532337,0,0,0,419.476,286.302,0.832952,392.057,322.853,0.838975,369.907,350.263,0.914812,444.305,286.319,0.861642,472.933,316.342,0.84146,449.473,350.239,0.909709,433.809,204.149,0.910381,442.906,204.149,0.953855,0,0,0,453.348,204.178,0.965901],"face_keypoints":[],"hand_left_keypoints":[],"hand_right_keypoints":[]}],"part_candidates":[{"0":[184.988,92.0557,0.160227,333.033,160.994,0.856558],"1":[340.007,168.992,0.881522],"2":[330.001,170.953,0.876429],"3":[354.979,185.007,0.98562],"4":[371.987,175,0.595711],"5":[352.01,165.009,0.819104],"6":[366.997,163.977,0.532337],"7":[366.016,163.014,0.86881],"8":[321.011,218.976,0.832952],"9":[300.003,246.98,0.838975],"10":[283.032,267.981,0.914812],"11":[340.035,218.989,0.861642],"12":[361.969,241.992,0.84146],"13":[343.994,267.963,0.909709],"14":[182.025,90.9823,0.171845,331.992,156.032,0.910381],"15":[187.036,90.9773,0.169177,338.962,156.032,0.953855],"16":[179.998,91.9771,0.0610613],"17":[191.986,93.0222,0.0871714,346.963,156.054,0.965901]}]}

### Type of Issue

- Execution error
- Help wanted
- Question

**Comment by gineshidalgo99** [comment]
Actually it's a scale issue (which I forgot to mention in the doc). `pose_keypoints` should be scaled to the output image resolution (unless you specify a different scale). However, `part_candidates` is the output of the deep net, so its resolution is in `net_resolution` scale. I'll put them into the same scale (and notify though this issue thread), but if you wanna make an immediate sense of them, simply rescale them

**Comment by DongJT1996** [comment]
ok, i get it. Thanks

**Comment by gineshidalgo99** [comment]
(I'll keep it open so I'll remember to do it and post here)

**Comment by DongJT1996** [comment]
oh, thanks

I just have a try to rescale the part_candidates. 
I do it as following:
if the size of input image is (491, 640, 3), and i use this command to get the json file:

./build/examples/openpose/openpose.bin --image_dir examples/media/input/ --part_candidates --write_json output/heatmaps/5/

Thus, i think the net_resolution is (368, 656). So if the coordinate of one point from the part_candidates is (184.988,92.0557), i think the rescaled point is (184.988 *491/368, 92.0557 *656/640). But when i plot the rescaled point, i find the y coordinate is wrong. 
Do i understand right?

**Comment by chrisdxie** [comment]
Any updates on this? I'd also like to use the part-candidates with the right resolution. Alternatively, is there a way to spit out the net resolution (when using "-1x368", it's not clear to me what the "-1" becomes)

**Comment by gineshidalgo99** [comment]
Hi all ( @DongJT1996  @chrisdxie ). The problem has been fixed in the latest commit which i've just pushed a few seconds ago. Now, part_candidates will always have the same scale than the keypoints themselves. Thanks for the feed-back!

**Comment by gineshidalgo99** [comment]
The body part candidates before being assembled into people (if `--part_candidates` is enabled). I.e., all the points that could be a body keypoint.
